HARJA-537 Sijainti pakolliseksi kun kirjataan levittäjällä tehtävää määrätoteumaa hoitourakalla

jarnovayrynen opened the pull request.

jarnovayrynen committed:
HARJA-537 Sijainti pakolliseksi levittäjätoteumalle hoitourakalla

Kun toteumalomakkeella syötetään levitettävää paikkausta, pakotetaan sijainti.

Korjataa tieosoitevalitsinta siten, että pakollisuutta indikoidaan oikein, riippuen vaaditaanko väli vai ei.

Lisää testidataan ne tehtävät, joille vaaditaan sijainti, jotta voidaan testata, käyttäen on conflict do nothing, koska tuotannossa ko. tehtävät jo ovat, ja syntyisi uniikkiusvaatimusta vastaan poikkeus.

jarnovayrynen committed:
Siirrä tehtävät testidataan, paranna tyylejä

jarnovayrynen committed:
Korjaa odotetut tehtävien lukumäärät testidatassa

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-537

jarnovayrynen committed:
Paranna aputekstiä kun sijainti on pakollinen

rckopteri commented on src/cljs/harja/tiedot/urakka/toteumat/maarien_toteumat.cljs:
```diff
@@ -67,6 +67,15 @@
                    ::t/lisatieto nil
                    ::t/maara nil})
 
+
```
Mietin vielä näiden sijaintia ja validointia. Onko tarpeen tarkistaa bäkärillä, että tietyille tehtäville toteumaa lisättäessä sijainti annetaan?

jarnovayrynen replied:
Joo, tehdäänpä backendiinkin tämä tarkistus. Aina varmempi.

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-537

jarnovayrynen committed:
Korjaa sijaintien validointia

Korjaa toteumalomaketta siten, että kun sijainti on pakollinen, se vaaditaan jotta tallennus on mahdollista. Lisäksi korjataan sulkeumavirhe, jotta pakollinen? -attribuutti menee perille asti kentät.cljs tierekisterikomponentissa, ja täten heijastelee tuoretta eikä alkutilaa.

jarnovayrynen committed:
Validoi määrätoteuman pakollinen sijainti myös palvelimella

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-537

jarnovayrynen committed:
Korjaa tr-flex kentän renderöintiä

Palauta sijaintien validointi määrätoteumalomakkeelle

jarnovayrynen committed:
Korjaa yksikkötestin onnistuva osa

jarnovayrynen committed:
Siirrä pääll.paikk. tehtävien luonti eri kohtaan

Vain testidataan muutos:
Jotta suunnitellut tehtävämäärät syntyvät, luodaan tehtävät ennen tehtävämäärien generointia.

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into HARJA-537
